google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1217,6 +1227,13 @@ def to_gapi
             @gapi
           end
 
+          def time_partitioning type: "DAY", expiration_ms: nil
```

Comment:
Works for me. And I agree that seconds should be sufficient precision for partitions of day length. The [API accepts milliseconds](https://cloud.google.com/bigquery/docs/creating-partitioned-tables#setting_the_expiration_of_a_partition) but we can convert when we write the API data.